### DISCUSSION TITLE
Enhancement request for more multi-select filterable fields site-wide

### DISCUSSION THREAD

**Discussion by StolltheStoll** [Heroes-Profile/heroesprofile#913] [Ideas]
**Describe the bug**
Not a bug, this is a very low priority enhancement request (and the last ticket I'm opening for today, I promise!).  I'm also aware that there's still some work being done on the filtering and data aggregation side, so this would be a far lower priority to that.

It isn't worth spending a lot of time on if it's structurally difficult, but if it's an easy change, it would be great if more filterable fields were multi-selects instead of single selects (eg. checkbox options instead of radio inputs).  Currently, the map filter field functions as a multi-select which is fantastic, but converting other fields would benefit some overlooked use cases:

- If a user wants to view their match history as a "Bruiser" and "Melee Assassin"
- If a user wants to view their match history as "Dehaka" and "Leoric"
- If a user wants to view their match history for 2023 Seasons 1 through 3

**To Reproduce**
Currently, since the Player Match History page has more filter criteria than others, this can be seen there, but as other features load into the site it will apply on other pages as well:
https://www.heroesprofile.com/Player/NovemberTera/11448173/1/Match/History

Heroes, Role, and Season filters only allow selections of 0 or 1 choice.

**Expected behavior**
Ideally, they would function more like the Map or Game Type filtering fields, where multiple options can be selected and there's the option to "Select All".  A blank input would ideally also implicitly act as the "Select All" choice (similar to how the Map field currently works.

Thanks for all the hard work on the new site Zemill!  It's looking great so far!

**Comment by Br1ghtw1ng** [comment]
Bump - i agree :)
